src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -459,6 +459,13 @@ private BoundExpression BindArgListOperator(InvocationExpressionSyntax node, Bin
                                     methodGroup.ResultKind);
                             break;
                     }
+
+                    if (receiver.Type.IsRefLikeType)
```

Comment:
This fixes the problem with invocations. Consider also addressing indexer accesses (I only tested ref struct receiver):
```
var s = new S();
dynamic d = null;

_ = s[d];

        ref struct S
{
    public int this[object o] => throw null;
}
```